.github/workflows/tests.yml

```diff
@@ -16,6 +22,13 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
+      - uses: actions/cache@v2
+        name: Configure pip caching
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/setup.py') }}
```

Comment:
> > hashFiles('**/setup.py') should probably be hashFiles('**/tox.ini') and maybe also cover the files in ./requirements.

Yep, I think the `tox.ini` file is fine. (Can't quite recall the last time requirements itself got bumped... 🙂)

Reply:
We want the hash to change when a dependency version gets bumped? 🤔 

The tox file will never change as tox installs the versions from the requirements file. So could get the hash of the text-ci.txt (and pin the dependencies in that file).

(either way, setup.py is wrong for this project)

Reply:
You were both way ahead of me on this. 

I've based the hash on the tox and the requirements file as these seem to be the two places items are installed from for the test run via tox. 

I guess the actual versions don't matter as pip will look in the cache and if it finds the right version it will use that, if not it will download the right version? 🤔  I can't pin the dependencies here as they vary by Python version. It's late here now, speak soon. 🛌